From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ppc64le build-scripts maintainers
Date: Mon, 01 Jan 2024 00:00:00 +0000
Subject: [PATCH] Exclude source that has commercial license (CC-BY-NC-4.0)

Remove SWAG model references (CC-BY-NC 4.0 license) from README and
exclude regnet and vision_transformer modules which contain SWAG
pre-trained weights from models/__init__.py.

diff --git a/README.md b/README.md
index 1234567..abcdefg 100644
--- a/README.md
+++ b/README.md
@@ -94,9 +94,6 @@ The pre-trained models provided in this library may have their own licenses or t
 dataset used for training. It is your responsibility to determine whether you have permission to use the models for your
 use case.
 
-More specifically, SWAG models are released under the CC-BY-NC 4.0 license. See
-[SWAG LICENSE](https://github.com/facebookresearch/SWAG/blob/main/LICENSE) for additional details.
-
 ## Citing TorchVision
 
 If you find TorchVision useful in your work, please consider citing the following BibTeX entry:
diff --git a/torchvision/models/__init__.py b/torchvision/models/__init__.py
index 1234567..abcdefg 100644
--- a/torchvision/models/__init__.py
+++ b/torchvision/models/__init__.py
@@ -6,12 +6,10 @@ from .googlenet import *
 from .inception import *
 from .mnasnet import *
 from .mobilenet import *
-from .regnet import *
 from .resnet import *
 from .shufflenetv2 import *
 from .squeezenet import *
 from .vgg import *
-from .vision_transformer import *
 from .swin_transformer import *
 from .maxvit import *
 from . import detection, optical_flow, quantization, segmentation, video
